pkg/daemon/config.go

```diff
@@ -92,3 +103,38 @@ func tryToLoadOldConfig(nodeProfilesJson []byte) ([]ptpv1.PtpProfile, bool) {
 
 	return []ptpv1.PtpProfile{*ptpConfig}, true
 }
+
+func populatePtp4lConf(config *string, output *ptp4lconf) {
+	lines := strings.Split(*config, "\n")
+	var current string;
+	output.sections = map[string]ptp4lconfSection{}
+
+	for _, line := range lines{
+		if strings.HasPrefix(line, "["){
+			current = line
+			output.sections[current] = *new(ptp4lconfSection)
+			if line != "[global]"{
+				output.ifaceString = fmt.Sprintf("%s%s", output.ifaceString, line)
+			}
+		} else if current != ""{
+			section := output.sections[current]
+			section.options = append(section.options, line)
+			output.sections[current] = section
+		}
+	}
+	output.ifaceString = strings.ReplaceAll(output.ifaceString, "][", ",")
+	output.ifaceString = strings.ReplaceAll(output.ifaceString, "[", "")
+	output.ifaceString = strings.ReplaceAll(output.ifaceString, "]", "")
+}
+
+func renderPtp4lConf(config *string, input *ptp4lconf) {
+
+	*config = ""
```

Comment:
same as before, there is no use having config as input and immediately assigning it ""
you could declare this func as 
```
func (p *ptp4lconf) Render() string { ...}
```

Reply:
config is the output when rendering the config based off of the input ptp4lconf.

Reply:
updated

Reply:
same maybe you didn't push? :)